scripts/Stretch.py

```diff
@@ -72,22 +72,27 @@ class Stretcher():
     """
     Execution part of the stretch algorithm
     """
-    def __init__(self, line_width, stretch):
+    def __init__(self, line_width, wc_stretch, pw_stretch):
         self.line_width = line_width
-        self.stretch = stretch
+        self.wc_stretch = wc_stretch
+        self.pw_stretch = pw_stretch
+        if self.pw_stretch > line_width / 4:
+            self.pw_stretch = line_width / 4 # Limit value of pushwall stretch distance
         self.outpos = GCodeStep(0)
         self.vd1 = np.empty((0, 2)) # Start points of segments
                                     # of already deposited material for current layer
         self.vd2 = np.empty((0, 2)) # End points of segments
                                     # of already deposited material for current layer
         self.layer_z = 0            # Z position of the extrusion moves of the current layer
+        self.layergcode = ""
 
     def execute(self, data):
         """
         Computes the new X and Y coordinates of all g-code steps
         """
-        Logger.log("d", "Post stretch with line width=" + str(self.line_width)
-                   + "mm and stretch=" + str(self.stretch)+ "mm")
+        Logger.log("d", "Post stretch with line width = " + str(self.line_width)
+                   + "mm wide circle stretch = " + str(self.wc_stretch)+ "mm"
+                   + "and push wall stretch = " + str(self.pw_stretch) + "mm")
```

Comment:
For this sort of thing we have formatting, for instance:

    Logger.log("d", "Post stretch with line width = {line_width}mm wide circle stretch = {wc_stretch}mm and push wall stretch = {pw_stretch}mm",
        line_width = self.line_width,
        wc_stretch = self.wc_stretch,
        pw_stretch = self.pw_stretch)